### PR TITLE
Use MB functions

### PR DESCRIPTION
**Pull request by DarkSide666**

### CODE REVIEW

File: src/Connection.php [hunk 1]
```diff
@@ -82,14 +82,14 @@ public static function normalizeDSN($dsn, $user = null, $pass = null)
 
         // If it's string, then find driver
         if (is_string($dsn)) {
-            if (strpos($dsn, ':') === false) {
+            if (mb_strpos($dsn, ':') === false) {
```
Comment (mvorisek) [inline src/Connection.php]: absolutely no need

File: src/Connection.php [hunk 2]
```diff
@@ -70,7 +70,7 @@ public static function normalizeDSN($dsn, $user = null, $pass = null)
             // DSN is using URL-like format, so we need to convert it
             $dsn = $parts['scheme'] . ':host=' . $parts['host']
                 . (isset($parts['port']) ? ';port=' . $parts['port'] : '')
-                . ';dbname=' . substr($parts['path'], 1);
+                . ';dbname=' . mb_substr($parts['path'], 1);
```
Comment (mvorisek) [inline src/Connection.php]: Probably to exclude `/` as the first char, so no need.

File: src/Expression.php [hunk 3]
```diff
@@ -587,7 +587,7 @@ public function execute(object $connection = null)
                     // workaround to support LOB data type 2/2, see https://github.com/doctrine/dbal/pull/2434
                     $statement->boundValues[$key] = $val;
                     if ($type === \PDO::PARAM_STR) {
-                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type, strlen((string) $val));
+                        $bind = $statement->bindParam($key, $statement->boundValues[$key], $type, mb_strlen((string) $val));
```
Comment (mvorisek) [inline src/Expression.php]: Be very carefull here, it may require a binary length here!!

File: src/Expression.php [hunk 4]
```diff
@@ -424,14 +424,14 @@ public function render()
             //     param     |  escape   |  escapeSoft
             '/\[[a-z0-9_]*\]|{[a-z0-9_]*}|{{[a-z0-9_]*}}/i',
             function ($matches) use (&$nameless_count) {
-                $identifier = substr($matches[0], 1, -1);
+                $identifier = mb_substr($matches[0], 1, -1);
```
Comment (mvorisek) [inline src/Expression.php]: Example of misleading usage - this strips brackets from the sides...

**Review by mvorisek — CHANGES_REQUESTED** [review]
Running CS fixer is fine, but you usually find 1, 2 valid usecases with these fixers... That is also why I reviewed the fixed files before applaying the most risky rules when added the global rules...

File: src/Expression.php (hunk 3 above)
Comment (DarkSide666) [reply src/Expression.php]: yes most likely it should be binary

File: src/Expression.php [hunk 5]
```diff
@@ -431,7 +431,7 @@ function ($matches) use (&$nameless_count) {
                 } elseif ($matches[0][0] === '{') {
                     if ($matches[0][1] === '{') {
                         $escaping = 'soft-escape';
-                        $identifier = substr($identifier, 1, -1);
+                        $identifier = mb_substr($identifier, 1, -1);
```
Comment (mvorisek) [inline src/Expression.php]: misleading only...

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: src/Expression.php (hunk 5 above)
Comment (DarkSide666) [reply src/Expression.php]: ok

**Comment by DarkSide666** [comment]
OK, nothing is actually left there. Only driverType which also should never be multibyte.
So I guess this PR can be closed without merging :)

File: src/Expression.php (hunk 3 above)
Comment (mvorisek) [reply src/Expression.php]: I do not know, should be checked, it is probably for maximal byte length to allocate, my guess.

**Pull request closed without merging**